ipwb/replay.py

```diff
@@ -179,6 +179,126 @@ def showMementosForURIRs_sansJS():
     return redirect('/memento/*/' + urir, code=301)
 
 
+def lookup_keys(surt):
+    keyre = re.compile(r"(.+)([,/]).+")
+    key = surt.split("?")[0].strip("/")
+    keys = [key, f"{key}/*"]
+    while "," in key:
+        try:
+            m = keyre.match(key)
+            keys.append(f"{m[1]}{m[2]}*")
+            key = m[1]
+        except Exception as e:
+            key = key.strip("/,")
+
+    return keys
+
+
+def bin_search(iter, key, datetime=None):
+    # Read line encompassing current position
+    ln = iter.readline()
+
+    while ipwbUtils.isCDXJMetadataRecord(ln):
+        ln = iter.readline()
+
+    surtk, rest = ln.split(maxsplit=1)
+    if key == surtk:
+        return [ln]
```

Comment:
If the first non-metadata line matches just the key, will you consider it a match?

* What about the datetime?
* What about the potential trailing slash in the surtk?

Reply:
Fixed in a603cf5.